execute/transport_test.go

```diff
@@ -162,3 +163,143 @@ func TestProcessChunkMsg(t *testing.T) {
 	}
 	mem.AssertSize(t, 0)
 }
+
+func TestWrapTransformationInTransport(t *testing.T) {
+	want := static.TableGroup{
+		static.StringKey("_m", "m0"),
+		static.StringKeys("t0", "a", "b"),
+		static.Times("_time", 0, 10, 20),
+		static.Floats("_value", 0, 1, 2),
+	}
+
+	t.Run("FlushKey", func(t *testing.T) {
+		var (
+			got      table.Iterator
+			finished bool
+		)
+
+		tr := execute.WrapTransformationInTransport(&mock.Transformation{
+			ProcessFn: func(id execute.DatasetID, tbl flux.Table) error {
+				buf, err := table.Copy(tbl)
+				if err != nil {
+					return err
+				}
+				got = append(got, buf)
+				return nil
+			},
+			FinishFn: func(id execute.DatasetID, err error) {
+				if err != nil {
+					t.Error(err)
+				}
+				finished = true
+			},
+		}, memory.DefaultAllocator)
+
+		processed := 0
+		if err := want.Do(func(tbl flux.Table) error {
+			if err := tbl.Do(func(cr flux.ColReader) error {
+				chunk := table.ChunkFromReader(cr)
+				chunk.Retain()
+				m := execute.NewProcessChunkMsg(chunk)
+				return tr.ProcessMessage(m)
+			}); err != nil {
+				return err
+			}
+
+			if got, want := len(got), processed; got != want {
+				t.Errorf("wrong number of tables processed -want/+got:\n\t- %d\n\t+ %d", want, got)
+			}
```

Comment:
I must be missing something here. At this point in the test, we have sent the process chunk message to `tr`, so I would expect the chunk to be copied and appended to `got`, giving it a length of 1. But we will not increment `processed` until line 217, so it will still be zero. How is this test passing?

Reply:
Ok, I get what my confusion was. The adapter won't actually invoke the legacy `Process` method until the flush key message is sent, because at that point it will know that it has a complete table. So `got` is not appended yet at this point.